WIP: Current working state

christianlandgraf opened the pull request.

sjahr commented on hybrid_planning_demo/include/hybrid_planning_demo/servo_solver.h:
```diff
@@ -0,0 +1,93 @@
+class ServoSolver : public moveit_hybrid_planning::LocalConstraintSolverInterface
+{
+public:
+  ServoSolver(){};
+  ~ServoSolver() override{};
```
I've removed the destructor definition from all plugin interface classes, so after pulling the latest changes from the hybrid planning branch this will fail.
```suggestion
  ServoSolver() = default;
  ~ServoSolver() = default;
```

henningkayser replied:
We can remove the constructor like in #37

sjahr commented on hybrid_planning_demo/include/hybrid_planning_demo/simple_servo_sampler.h:
```diff
@@ -0,0 +1,69 @@
+class SimpleServoSampler : public moveit_hybrid_planning::TrajectoryOperatorInterface
+{
+public:
+  SimpleServoSampler(){};
+  ~SimpleServoSampler() override{};
+
```
Same as above
```suggestion
  SimpleServoSampler() = default;
  ~SimpleServoSampler() = default;

```

henningkayser commented on .pre-commit-config.yaml:
```diff
@@ -17,7 +17,7 @@ repos:
     rev: v3.4.0
     hooks:
-      - id: check-added-large-files
+      # - id: check-added-large-files
```
Ideally, we would use git-lfs for large files

henningkayser commented on hybrid_planning_demo/src/global_mtc_planner.cpp:
```diff
@@ -69,36 +118,173 @@ GlobalMTCPlannerComponent::plan(const moveit_msgs::msg::MotionPlanRequest& plann
+    goal_pose.pose.position.x = motion_plan_req.goal_constraints[0].position_constraints[0].target_point_offset.x;
+    goal_pose.pose.position.y = motion_plan_req.goal_constraints[0].position_constraints[0].target_point_offset.y;
+    goal_pose.pose.position.z = motion_plan_req.goal_constraints[0].position_constraints[0].target_point_offset.z;
+    goal_pose.pose.orientation.x = motion_plan_req.goal_constraints[0].orientation_constraints[0].orientation.x;
```
```suggestion
    goal_pose.pose.orientation = motion_plan_req.goal_constraints[0].orientation_constraints[0].orientation;
```
remove x, y, z, w

henningkayser commented on hybrid_planning_demo/src/global_mtc_planner.cpp:
```diff
@@ -69,36 +118,173 @@ GlobalMTCPlannerComponent::plan(const moveit_msgs::msg::MotionPlanRequest& plann
+    goal_pose.pose.position.x = motion_plan_req.goal_constraints[0].position_constraints[1].target_point_offset.x;
+    goal_pose.pose.position.y = motion_plan_req.goal_constraints[0].position_constraints[1].target_point_offset.y;
+    goal_pose.pose.position.z = motion_plan_req.goal_constraints[0].position_constraints[1].target_point_offset.z;
+    goal_pose.pose.orientation.x = motion_plan_req.goal_constraints[0].orientation_constraints[1].orientation.x;
```
same

christianlandgraf replied:
I had trouble changing this since target_point_offset is of type geometry_msgs/Vector3d and not geometry_msgs/Point

henningkayser commented on hybrid_planning_demo/src/global_mtc_planner.cpp:
```diff
@@ -69,36 +118,173 @@ GlobalMTCPlannerComponent::plan(const moveit_msgs::msg::MotionPlanRequest& plann
+    t.add(std::move(stage));
+  }
+
+  /******************************************************
```
I think we can remove unused motions

henningkayser reviewed (commented):
Wow, this is so much new code. I'm not sure if I can really review all of this in depth, but I will test this and provide fixup where adequate. There are definitely some things that could be improved, but I don't think we should go over board with optimizations for this demo (yet). Could you resolve the conflicts with master?

christianlandgraf commented:
> 
> 
> Wow, this is so much new code. I'm not sure if I can really review all of this in depth, but I will test this and provide fixup where adequate. There are definitely some things that could be improved, but I don't think we should go over board with optimizations for this demo (yet). Could you resolve the conflicts with master?

Thanks for having a look at it! I think you can skip most of the processit_*/... stuff since it is not focus of the demo and needs a lot of improvement anyway :D
I will resolve the conflict and try to give a better description in the PR today

christianlandgraf commented:
I addressed the first reviews and fixed a minor[ bug(?) ](https://github.com/ros-planning/moveit_task_constructor/pull/170#discussion_r719602378)with the MTC namespace.
